docs/SPECS.md

```diff
@@ -0,0 +1,1435 @@
+      * [Deal Phase](#deal-phase)
+      * [Response Phase](#response-phase)
+      * [Justification Phase](#justification-phase)
+      * [Finish Phase](#finish-phase)
+   * [Randomness generation](#randomness-generation)
+      * [Overiew](#overiew)
+      * [Randomness Generation Period](#randomness-generation-period)
+      * [Beacon Chain](#beacon-chain)
+      * [Catchup mode](#catchup-mode)
+      * [Syncing](#syncing)
+* [Cryptographic specification](#cryptographic-specification)
+   * [Drand Curve](#drand-curve)
+   * [Distributed Public Key](#distributed-public-key)
+   * [Beacon Signature](#beacon-signature)
+   * [Partial Beacon Signature](#partial-beacon-signature)
+   * [Distributed Key Generation](#distributed-key-generation-1)
+      * [Input](#input)
+      * [Authentication](#authentication)
+      * [Setup](#setup)
+      * [Deal Phase](#deal-phase-1)
+      * [Response Phase](#response-phase-1)
+      * [Justification Phase](#justification-phase-1)
+      * [Finish Phase](#finish-phase-1)
+   * [Resharing](#resharing)
+      * [Setup](#setup-1)
+      * [Distinction of Roles](#distinction-of-roles)
+      * [Deal Phase](#deal-phase-2)
+      * [Response Phase](#response-phase-2)
+      * [Justification Phase](#justification-phase-2)
+      * [Finish Phase](#finish-phase-2)
+* [External API](#external-api)
+* [Control API](#control-api)
+* [Appendix A. DKG packets](#appendix-a-dkg-packets)
+* [THINGS TO REVIEW](#things-to-review)
+
+## Notations
+
+### Drand node
+
+A drand node is a server that runs the drand code, that participates in the
+distributed key generations phases, in the randomness generation and that can
+reply to public request API. A Go struct representation is as follow:
+```go
+type Node struct {
+	Key  []byte // public key on bls12-381 G1 
+	Addr string // publicly reachable address of the node
+	TLS  bool // reachable via TLS 
+    Index  uint32 // index of the node w.r.t. to the network
+}
+```
+A node can be referenced by its hash as follows:
+```go
+func (n *Node) Hash() []byte {
+    h := blake2b.New(nil)
+    h.Write([]byte(n.Addr))
+	binary.Write(h, binary.LittleEndian, n.Index)
+    h.Write(n.Key)
+    return h.Sum(nil)
+}
+```
+
+**Public Key**: Public keys of drand nodes are points on the G1 group of the
+BLS12-381 curve. See the [curve](#drand-curve) section for more information.
+
+### Drand beacon
+
+A drand beacon is what the drand network periodically creates and that can be
+used to derive the randomness. A beacon contains the signature of the previous
+beacon generated, the round of this beacon and signature. See the [beacon
+chain](#beacon-chain) section for more information.
+
+### Group configuration
+
+Group configuration: A structure that contains all the necessary information
+about nodes that form a drand network:
+* Nodes: A list of nodes information that represents all nodes on the network
+* Threshold: The number of nodes that are necessary to participate to a
```

Comment:
can we mention the bounds for this parameter under a crash-stop and byzantine failure model?